### PR TITLE
Package vdom.0.3

### PR DESCRIPTION
**Pull request by AurelienSaue**
### `vdom.0.3`
DOM and VDOM for OCaml
This package contains:
- OCaml bindings to DOM and other client-side Javascript APIs (using gen_js_api).
- An implementation of the Elm architecture, where the UI is specified as a functional "view" on the current state.

---
* Homepage: https://github.com/LexiFi/ocaml-vdom
* Source repo: git+https://github.com/LexiFi/ocaml-vdom.git
* Bug tracker: https://github.com/LexiFi/ocaml-vdom/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/vdom/vdom.0.3/opam [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+maintainer: ["Alain Frisch <alain.frisch@lexifi.com>"]
+authors: [
+  "Alain Frisch <alain.frisch@lexifi.com>"
+  "Marc Lasson <marc.lasson@lexifi.com>"
+  "Aurélien Saue <aurelien.saue@lexifi.com>"
+]
+homepage: "https://github.com/LexiFi/ocaml-vdom"
+license: "MIT"
+bug-reports: "https://github.com/LexiFi/ocaml-vdom/issues"
+depends: [
```
Comment (mseri) [inline packages/vdom/vdom.0.3/opam:17]:
```suggestion
depends: [
  "ocaml" {>= "4.10"}
```
This is where `concat_map` was introduced:
```
#=== ERROR while compiling vdom.0.3 ===========================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/LexiFi/ocaml-vdom/archive/refs/tags/0.3.tar.gz)
# path                 ~/.opam/4.08/.opam-switch/build/vdom.0.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p vdom -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/vdom-7-475ce6.env
# output-file          ~/.opam/log/vdom-7-475ce6.out
### output ###
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.vdom_js.objs/byte -I /home/opam/.opam/4.08/lib/js_of_ocaml-compiler/runtime -I /home/opam/.opam/4.08/lib/ojs -I lib/.vdom_base.objs/byte -intf-suffix .ml -no-alias-deps -o lib/.vdom_js.objs/byte/vdom_blit.cmo -c -impl lib/vdom_blit.ml)
# File "lib/vdom_blit.ml", line 506, characters 17-32:
# 506 |       let doms = List.concat_map get_doms children in
#                        ^^^^^^^^^^^^^^^
# Error: Unbound value List.concat_map
```

File: packages/vdom/vdom.0.3/opam [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  "dune" {>= "3.0"}
+  "gen_js_api" {>= "1.0.6"}
+  "js_of_ocaml-compiler"
```
Comment (mseri) [inline packages/vdom/vdom.0.3/opam:20]:
```suggestion
  "ojs"
  "js_of_ocaml-compiler"
```
due to
```

#=== ERROR while compiling vdom.0.3 ===========================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.10.2 | pinned(https://github.com/LexiFi/ocaml-vdom/archive/refs/tags/0.3.tar.gz)
# path                 ~/.opam/4.10/.opam-switch/build/vdom.0.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p vdom -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/vdom-7-a5523d.env
# output-file          ~/.opam/log/vdom-7-a5523d.out
### output ###
# File "lib/dune", line 17, characters 12-15:
# 17 |  (libraries ojs vdom_base)
#                  ^^^
# Error: Library "ojs" not found.
# -> required by library "vdom" in _build/default/lib
# -> required by _build/default/META.vdom
# -> required by _build/install/default/lib/vdom/META
# -> required by _build/default/vdom.install
# -> required by alias install
```

Comment (AurelienSaue) [reply packages/vdom/vdom.0.3/opam:22]: I think it is better to require gen_js_api >= 1.0.7

Comment (mseri) [reply packages/vdom/vdom.0.3/opam:22]: Feel free to update the bound, but the package explicitly links against `ojs` so it should be there as a dependency

**Comment by mseri** [comment]
No need, the released package will use the opam-repository metadata as the source of truth, it is enough that you update your main branch with the changes.

**Comment by AurelienSaue** [comment]
It looks like there is an issue with gen_js_api.1.0.7's lower bound on ocaml-migrate-parsetree.

**Comment by mseri** [comment]
That’s something we can fix separately and does not depend on your pr. Thanks!

feel free to announce the release on discuss.ocaml.org!

File: packages/vdom/vdom.0.3/opam [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+homepage: "https://github.com/LexiFi/ocaml-vdom"
+license: "MIT"
+bug-reports: "https://github.com/LexiFi/ocaml-vdom/issues"
+depends: [
+  "ocaml" {>= "4.10"}
+  "dune" {>= "3.0"}
+  "gen_js_api" {>= "1.0.7"}
+  "ojs"
+  "js_of_ocaml-compiler"
+  "odoc"
```
Comment (copy) [inline packages/vdom/vdom.0.3/opam:23]: @AurelienSaue Does vdom really depend odoc?

Comment (mseri) [reply packages/vdom/vdom.0.3/opam:23]: It does not seem the case, however it was done intentionally: https://github.com/LexiFi/ocaml-vdom/pull/43

Comment (AurelienSaue) [reply packages/vdom/vdom.0.3/opam:23]: It doesn't, the `{with-doc}` got lost. My apologies!

Comment (AurelienSaue) [reply packages/vdom/vdom.0.3/opam:23]: @mseri should I PR the fix?

Comment (mseri) [reply packages/vdom/vdom.0.3/opam:23]: @AurelienSaue no thanks, I have just done it here: https://github.com/ocaml/opam-repository/pull/24713
Please make a PR to fix the issue upstream if it is still there

Comment (AurelienSaue) [reply packages/vdom/vdom.0.3/opam:23]: I did, thank you!

**Pull request merged**